main.cpp

```diff
@@ -0,0 +1,419 @@
+#include <SFML/Graphics.hpp>
+#include <SFML/Network.hpp>
+#include <iostream>
+#include <windows.h>
+#include <map>
+#include "TextureHolder.h"
+#include "Square.h"
+#include "Piece.h"
+
+bool move(Square* slctd, Square* trgt, sf::UdpSocket& sckt, sf::IpAddress& IP, unsigned short& PORT, sf::Packet& pck);
+
+int main()
+{	
+	//kod debugowania = 1337
+	bool turn=false, gamestarted=false, iswhite = true;
+	sf::Packet pack;
+	sf::UdpSocket socket;
+	sf::IpAddress serveradress;
+	unsigned short serverport;
+	int serverconnectionstatus = 1, id, ctx, cty;
+	float tx, ty; //koordy targetu do update
+
+
+
+
+	std::cout << "Server IP: ";
+	std::cin >> serveradress;
+	std::cout << "Server Port: ";
+	std::cin >> serverport;
+
+
+	if (serverport != 1337) { // debug - wpisa� w pole "port" 1337 aby przejsc do inicjalizacji gry
+		if (socket.bind(socket.getLocalPort()) != sf::Socket::Done) {
+			std::cout << "Failed to bind a socket!" << std::endl;
+			exit(4);
+		}
+
+		pack << serverconnectionstatus;
+		if (socket.send(pack, serveradress, serverport) != sf::Socket::Done) {
+			std::cout << "Failed to send connection initialization packet!" << std::endl;
+			exit(2);
+		}
+		pack.clear();
+		if (socket.receive(pack, serveradress, serverport) != sf::Socket::Done) {
+			std::cout << "Failed to recieve initialization packet!" << std::endl;
+			exit(2);
+		}
+		serverconnectionstatus = 0;
+		pack >> serverconnectionstatus >> iswhite;
+		std::cout << "CONNSTATUS: " << serverconnectionstatus << std::endl;
+		std::cout << std::boolalpha;
+		std::cout << "Is white? " << iswhite << std::endl;
+
+		if (serverconnectionstatus != 2) {
+			std::cout << "Connection not stablilized!" << std::endl;
+			exit(3);
+		}
+
+		std::cout << "Waiting for 2nd player" << std::endl;
+		pack.clear();
+		if (socket.receive(pack, serveradress, serverport) != sf::Socket::Done) {
+			std::cout << "Failed to recieve initialization packet!" << std::endl;
+			exit(2);
+		}
+		pack >> gamestarted;
+		if (gamestarted) 
+			std::cout << "Everything is ready!" << std::endl;
+
+		else {
+			std::cout << "2nd player didn't come, terminating..." << std::endl;
+			exit(69);
+		}
+
+}
+	
+	//---------------------------------------------------------------------------------------------------------
```

Comment:
...
nie lepiej oddzielną funkcję zrobić od takich fresków?
Jeżeli zmienne w main były by memberami jakiejś klasy, to nie było by problemu z ich przekazywaniem.